### PR TITLE
Upgrade to Ruby 3.1.2 & rails 7

### PR DESCRIPTION
**Pull request by CloCkWeRX**
- [x] Rails 7? Something else?
- [x] ```DEPRECATION WARNING: TimeWithZone#to_s(:rfc822) is deprecated. Please use TimeWithZone#to_fs(:rfc822) instead. (called from block (2 levels) in <top (required)> at /home/runner/work/growstuff/growstuff/spec/views/members/show.rss.haml_spec.rb:13)```
- [x] ```DEPRECATION WARNING: Time#to_s(:number) is deprecated. Please use Time#to_fs(:number) instead. (called from cache_key_for at /home/runner/work/growstuff/growstuff/app/helpers/application_helper.rb:27)```
- [x] Yell at codeclimate
- [x] Whatever the 5x last deprecation warnings are (Now errors)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ source 'https://rubygems.org'
 # Match ruby version in .ruby-version
 ruby File.read('.ruby-version')
 
-gem 'rails', '~> 6.1.0'
+gem 'rails', '~> 7.0.0'
```
Comment (Br3nda) [inline Gemfile:8]: amazing!

Comment (CloCkWeRX) [reply Gemfile:8]: @cesy I think... I did a bunch of get rid of deprecations some time ago; which is why it's a bit cleaner.

Might go dependabot's version over this; but for buildpack-22..

**Comment by CloCkWeRX** [comment]
If anyone wants to go through and close off the Brakeman "notes" where they aren't relevant, that'd be a help.
Alternatively, pushing attr_accessible whitelisting for the last 3x models.

**Comment by CloCkWeRX** [comment]
Oh, far out. attr_accessible doesn't exist :/ 
Thanks, Brakeman.

**Comment by cesy** [comment]
Thanks for all your work on this. Looks like once we've got Ruby upgraded, we can then get onto a newer version of the Heroku stack. One more test failure I can see here.

**Comment by CloCkWeRX** [comment]
Todo: revert https://github.com/Growstuff/growstuff/pull/3031/commits/fe54eee35cc1e40b811bae4fe3805c5eef22e856 / the haml update

**Comment by CloCkWeRX** [comment]
Let's see how this last PR run goes. Off to sleep for me.

**Comment by CloCkWeRX** [comment]
Have pushed this onto staging, mostly trusting the CI results; not much local test at all.

**Comment by Br3nda** [comment]
27 fixed issues!

**Comment by CloCkWeRX** [comment]
Going to merge this, did a small amount of regression testing on mobile.
Only issues I saw were smtp related.

**Pull request merged**